source/agora/consensus/validation/Block.d

```diff
@@ -615,10 +622,26 @@ unittest
     foreach (ref tx; GenesisBlock.txs)
         utxo_set.put(tx);
 
-    auto txs_1 = makeChainedTransactions(gen_key, null, 1,
-        400_000_000_000 * 8).sort.array;
+    auto txs_1 = genesisSpendable().map!(txb => txb.sign()).array();
 
     auto block1 = makeNewBlock(GenesisBlock, txs_1);
+
+    foreach (idx1, txs1; block1.txs)
+    {
+        foreach (idx2, txs2; block1.txs[idx1].outputs)
+            block1.txs[idx1].outputs[idx2].value = Amount(400_000_000_000);
+    }
+
+    foreach (idx1, txs1; block1.txs)
+    {
+        foreach (idx2, txs2; block1.txs[idx1].inputs)
+            block1.txs[idx1].inputs[idx2].signature = gen_key.secret.sign(
+                hashFull(block1.txs[idx1])[]);
+    }
```

Comment:
The code could be simpler like as follows.

```suggestion
    foreach (tx; block1.txs)
    {
        iota(tx.outputs.length).map!(
            idx => tx.outputs[idx].value = Amount(400_000_000_000));
        iota(tx.inputs.length).map!(
            idx => tx.inputs[idx].signature = gen_key.secret.sign(hashFull(tx)[]));
    }
```

Reply:
Thank you for your suggestion! I will keep it in mind for future `foreach` loops.
However, this part will be deleted because the `Amount` does not matter for this test.